Source/WebCore/Modules/mediastream/PeerConnectionBackend.cpp

```diff
@@ -129,10 +139,47 @@ void PeerConnectionBackend::createOffer(RTCOfferOptions&& options, CreateCallbac
     doCreateOffer(WTFMove(options));
 }
 
+#if !RELEASE_LOG_DISABLED
+bool PeerConnectionBackend::handleLogMessage(const WTFLogChannel& channel, WTFLogLevel, Vector<JSONLogValue>&& values)
+{
+    auto name = StringView::fromLatin1(channel.name);
+    if (name != "WebRTC"_s)
+        return false;
+
+    // Ignore logs containing only the call site information or JSON logs.
+    if (values.size() < 2 || values[1].type == JSONLogValue::Type::JSON)
+        return false;
+
+    Ref peerConnection = m_peerConnection.get();
+    auto document = peerConnection->document();
+    if (!document)
+        return false;
+
+    auto page = document->page();
```

Comment:
This triggers an assert because can be called from a non-main thread. I'll see what i can do about this...